src/tests/catwalk_tests/test_storage.py

```diff
@@ -174,6 +177,7 @@ def test_MatrixStore_labels_idempotency():
 def test_MatrixStore_save():
     data = {
         "entity_id": [1, 2],
+        "as_of_date": [pd.Timestamp(2017, 1, 1), pd.Timestamp(2017, 1, 1)],
```

Comment:
I've seen both `pd.Timestamp` and `datetime.date` being used in triage. I know we were having some problem with pandas using `datetime`. Just curious when to use `pd.Timestamp` and when to use `datetime`.

Reply:
Pandas automatically converts to `pd.Timestamp` when it parses dates, so when we are working with matrix data like here I think pd.Timestamp should be used to reflect what will happen in reality. 

Metadata can use `datetime.date` (and indeed the MatrixStore converts to date when it returns as-of-dates from the metadata) for ease of use (you can have a list of dates and compare them).

So the dividing line between `pd.Timestamp` and `datetime.date` is whether or not it is data in a pandas dataframe or not. Maybe we can do better, but I think standardizing either way may be a bit harder :man_shrugging:

Reply:
I agree,  even though they claim that `pd.Timestamp` and `datetime.date` are interchangeable in **most** cases   😒.